--- cairo-1.10.0/build/configure.ac.warnings.orig	2020-07-13 12:22:37.920663090 +0200
+++ cairo-1.10.0/build/configure.ac.warnings	2020-07-13 12:23:44.624573762 +0200
@@ -27,7 +27,7 @@
 
 dnl We also abuse the warning-flag facility to enable other compiler
 dnl options.  Namely, the following:
-MAYBE_WARN="$MAYBE_WARN -fno-strict-aliasing -fno-common -flto"
+MAYBE_WARN="$MAYBE_WARN -fno-strict-aliasing -fno-common"
 
 dnl Also to turn various gcc/glibc-specific preprocessor checks
 MAYBE_WARN="$MAYBE_WARN -Wp,-D_FORTIFY_SOURCE=2"
