diff -burN orig.cairo-1.10.0/build/config.sub cairo-1.10.0/build/config.sub
--- orig.cairo-1.10.0/build/config.sub	2010-03-29 08:21:24.000000000 -0300
+++ cairo-1.10.0/build/config.sub	2010-11-03 16:15:41.541875160 -0300
@@ -4,7 +4,7 @@
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009
 #   Free Software Foundation, Inc.
 
-timestamp='2009-11-20'
+timestamp='2010-11-03'
 
 # This file is (in principle) common to ALL GNU software.
 # The presence of a machine in this file suggests that SOME GNU software
@@ -941,6 +941,10 @@
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+	        basic_machine=powerpc64-unknown
+		os=-linux
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
