diff -burN orig.freetype-2.4.3/builds/unix/config.sub freetype-2.4.3/builds/unix/config.sub
--- orig.freetype-2.4.3/builds/unix/config.sub	2010-10-03 15:45:06.000000000 -0300
+++ freetype-2.4.3/builds/unix/config.sub	2010-11-03 14:52:11.751873859 -0300
@@ -4,7 +4,7 @@
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010
 #   Free Software Foundation, Inc.
 
-timestamp='2010-09-11'
+timestamp='2010-11-03'
 
 # This file is (in principle) common to ALL GNU software.
 # The presence of a machine in this file suggests that SOME GNU software
@@ -970,6 +970,10 @@
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+	        basic_machine=powerpc64-unknown
+		os=-linux
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
