diff -burN orig.jpeg-8b/config.sub jpeg-8b/config.sub
--- orig.jpeg-8b/config.sub	2010-05-02 13:49:39.000000000 -0300
+++ jpeg-8b/config.sub	2010-11-03 14:56:27.401875133 -0300
@@ -4,7 +4,7 @@
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009
 #   Free Software Foundation, Inc.
 
-timestamp='2009-11-20'
+timestamp='2010-11-03'
 
 # This file is (in principle) common to ALL GNU software.
 # The presence of a machine in this file suggests that SOME GNU software
@@ -941,6 +941,10 @@
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+	        basic_machine=powerpc64-unknown
+		os=-linux
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
