diff -burN orig.libogg-1.2.1/config.sub libogg-1.2.1/config.sub
--- orig.libogg-1.2.1/config.sub	2006-08-10 12:17:12.000000000 -0300
+++ libogg-1.2.1/config.sub	2010-11-04 14:31:44.071876933 -0300
@@ -4,7 +4,7 @@
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006 Free Software Foundation,
 #   Inc.
 
-timestamp='2006-07-02'
+timestamp='2010-11-04'
 
 # This file is (in principle) common to ALL GNU software.
 # The presence of a machine in this file suggests that SOME GNU software
@@ -872,6 +872,10 @@
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+	        basic_machine=powerpc64-unknown
+		os=-linux
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
