diff -burN orig.pixman-0.20.0/config.sub pixman-0.20.0/config.sub
--- orig.pixman-0.20.0/config.sub	2009-10-15 09:05:46.000000000 -0300
+++ pixman-0.20.0/config.sub	2010-11-04 13:38:14.195624946 -0300
@@ -4,7 +4,7 @@
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008
 #   Free Software Foundation, Inc.
 
-timestamp='2009-04-17'
+timestamp='2010-11-04'
 
 # This file is (in principle) common to ALL GNU software.
 # The presence of a machine in this file suggests that SOME GNU software
@@ -925,6 +925,10 @@
 	ppc64le-* | powerpc64little-*)
 		basic_machine=powerpc64le-`echo $basic_machine | sed 's/^[^-]*-//'`
 		;;
+	ppu)
+	        basic_machine=powerpc64-unknown
+		os=-linux
+		;;
 	ps2)
 		basic_machine=i386-ibm
 		;;
