diff -Nru SDL-1.2.15/configure.in SDL-1.2.15/configure.in
--- SDL-1.2.15/configure.in	2012-01-20 22:49:48
+++ SDL-1.2.15/configure.in	2026-09-05 20:25:36
@@ -2805,6 +2805,19 @@
             have_timers=yes
         fi
         ;;
+    powerpc64-ps3-elf*)
+        ARCH=psl1ght
+        EXTRA_CFLAGS="$EXTRA_CFLAGS -D__PSL1GHT__ -DHAVE_POW -DHAVE_MMAP=0"
+        EXTRA_CFLAGS="$EXTRA_CFLAGS -I\"$PSL1GHT/ppu/include\" -mcpu=cell"
+        EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lrt -llv2"
+        CheckDummyVideo
+        CheckDummyAudio
+        if test x$enable_timers = xyes; then
+            AC_DEFINE(SDL_TIMER_PSL1GHT)
+            SOURCES="$SOURCES $srcdir/src/timer/psl1ght/*.c"
+            have_timers=yes
+        fi
+        ;;
     *-riscos)
         ARCH=riscos
         CheckOSS
diff -Nru SDL-1.2.15/include/SDL_config.h.in SDL-1.2.15/include/SDL_config.h.in
--- SDL-1.2.15/include/SDL_config.h.in	2012-01-20 22:49:48
+++ SDL-1.2.15/include/SDL_config.h.in	2026-09-05 20:25:36
@@ -249,6 +249,7 @@
 #undef SDL_TIMER_OS2
 #undef SDL_TIMER_RISCOS
 #undef SDL_TIMER_UNIX
+#undef SDL_TIMER_PSL1GHT
 #undef SDL_TIMER_WIN32
 #undef SDL_TIMER_WINCE
 
diff -Nru SDL-1.2.15/include/SDL_platform.h SDL-1.2.15/include/SDL_platform.h
--- SDL-1.2.15/include/SDL_platform.h	2012-01-20 22:49:48
+++ SDL-1.2.15/include/SDL_platform.h	2026-09-05 20:25:36
@@ -86,6 +86,10 @@
 #undef __OS2__
 #define __OS2__		1
 #endif
+#if defined(__PS3__) || defined(__PSL1GHT__)
+#undef __PS3__
+#define __PS3__		1
+#endif
 #if defined(osf) || defined(__osf) || defined(__osf__) || defined(_OSF_SOURCE)
 #undef __OSF__
 #define __OSF__		1
diff -Nru SDL-1.2.15/src/timer/psl1ght/SDL_systimer.c SDL-1.2.15/src/timer/psl1ght/SDL_systimer.c
--- SDL-1.2.15/src/timer/psl1ght/SDL_systimer.c	1969-12-31 21:00:00
+++ SDL-1.2.15/src/timer/psl1ght/SDL_systimer.c	2026-09-05 20:25:36
@@ -0,0 +1,49 @@
+#include "SDL_config.h"
+
+#ifdef SDL_TIMER_PSL1GHT
+
+#include <sys/time.h>
+#include <unistd.h>
+
+#include "SDL_timer.h"
+#include "../SDL_timer_c.h"
+
+static struct timeval start;
+
+void SDL_StartTicks(void)
+{
+	gettimeofday(&start, NULL);
+}
+
+Uint32 SDL_GetTicks(void)
+{
+	struct timeval now;
+	gettimeofday(&now, NULL);
+	return (Uint32)((now.tv_sec - start.tv_sec) * 1000 + (now.tv_usec - start.tv_usec) / 1000);
+}
+
+void SDL_Delay(Uint32 ms)
+{
+	usleep(ms * 1000);
+}
+
+int SDL_SYS_TimerInit(void)
+{
+	return 0;
+}
+
+void SDL_SYS_TimerQuit(void)
+{
+}
+
+int SDL_SYS_StartTimer(void)
+{
+	SDL_SetError("Timers are not supported on this platform");
+	return -1;
+}
+
+void SDL_SYS_StopTimer(void)
+{
+}
+
+#endif /* SDL_TIMER_PSL1GHT */
