diff -burN tiff-3.9.4.orig//contrib/iptcutil/iptcutil.c tiff-3.9.4/contrib/iptcutil/iptcutil.c
--- tiff-3.9.4.orig//contrib/iptcutil/iptcutil.c	2011-01-30 15:01:15.759211660 +0100
+++ tiff-3.9.4/contrib/iptcutil/iptcutil.c	2011-01-30 15:01:21.274911484 +0100
@@ -5,7 +5,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <memory.h>
 #include <ctype.h>
 
 #ifdef HAVE_STRINGS_H
