diff -burN orig.zlib-1.2.6/Makefile.in zlib-1.2.6/Makefile.in
--- orig.zlib-1.2.6/Makefile.in	2012-01-28 23:48:50.000000000 +0100
+++ zlib-1.2.6/Makefile.in	2012-02-17 19:12:07.109166356 +0100
@@ -73,7 +73,7 @@
 
 all: static shared
 
-static: example$(EXE) minigzip$(EXE)
+static: $(STATICLIB) # example$(EXE) minigzip$(EXE)
 
 shared: examplesh$(EXE) minigzipsh$(EXE)
 
diff -burN orig.zlib-1.2.6/configure zlib-1.2.6/configure
--- orig.zlib-1.2.6/configure	2012-01-29 18:48:11.000000000 +0100
+++ zlib-1.2.6/configure	2012-02-17 19:14:34.119121726 +0100
@@ -182,8 +182,8 @@
   MINGW* | mingw*)
 # temporary bypass
         rm -f $test.[co] $test $test$shared_ext
-        echo "Please use win32/Makefile.gcc instead." | tee -a configure.log
-        exit 1
+#        echo "Please use win32/Makefile.gcc instead." | tee -a configure.log
+#        exit 1
         LDSHARED=${LDSHARED-"$cc -shared"}
         LDSHAREDLIBC=""
         EXE='.exe' ;;
@@ -231,8 +231,7 @@
              SHAREDLIBV=libz.$VER$shared_ext
              SHAREDLIBM=libz.$VER1$shared_ext
              LDSHARED=${LDSHARED-"$cc -dynamiclib -install_name $libdir/$SHAREDLIBM -compatibility_version $VER1 -current_version $VER3"}
-             AR="/usr/bin/libtool"
-             ARFLAGS="-o" ;;
+      ;;
   *)             LDSHARED=${LDSHARED-"$cc -shared"} ;;
   esac
 else
