--- orig.zlib-1.2.11/Makefile.in	2017-01-15 18:29:40.000000000 +0100
+++ zlib-1.2.11/Makefile.in	2017-02-15 12:02:49.221883981 +0100
@@ -75,7 +75,7 @@
 
 all: static shared
 
-static: example$(EXE) minigzip$(EXE)
+static: $(STATICLIB) # example$(EXE) minigzip$(EXE)
 
 shared: examplesh$(EXE) minigzipsh$(EXE)
 
--- orig.zlib-1.2.11/configure	2016-12-31 19:06:40.000000000 +0100
+++ zlib-1.2.11/configure	2017-02-15 12:02:49.244443762 +0100
@@ -217,8 +217,8 @@
   MINGW* | mingw*)
 # temporary bypass
         rm -f $test.[co] $test $test$shared_ext
-        echo "Please use win32/Makefile.gcc instead." | tee -a configure.log
-        leave 1
+        # echo "Please use win32/Makefile.gcc instead." | tee -a configure.log
+        # leave 1
         LDSHARED=${LDSHARED-"$cc -shared"}
         LDSHAREDLIBC=""
         EXE='.exe' ;;
@@ -241,12 +241,7 @@
              SHAREDLIBV=libz.$VER$shared_ext
              SHAREDLIBM=libz.$VER1$shared_ext
              LDSHARED=${LDSHARED-"$cc -dynamiclib -install_name $libdir/$SHAREDLIBM -compatibility_version $VER1 -current_version $VER3"}
-             if libtool -V 2>&1 | grep Apple > /dev/null; then
-                 AR="libtool"
-             else
-                 AR="/usr/bin/libtool"
-             fi
-             ARFLAGS="-o" ;;
+             ;;
   *)             LDSHARED=${LDSHARED-"$cc -shared"} ;;
   esac
 else
