diff -burN orig.zlib-1.2.5/Makefile.in zlib-1.2.5/Makefile.in
--- orig.zlib-1.2.5/Makefile.in	2010-04-20 00:12:21 -0400
+++ zlib-1.2.5/Makefile.in	2011-03-06 00:58:14 -0500
@@ -69,7 +69,7 @@
 
 all: static shared
 
-static: example$(EXE) minigzip$(EXE)
+static: $(STATICLIB) # example$(EXE) minigzip$(EXE)
 
 shared: examplesh$(EXE) minigzipsh$(EXE)
 
@@ -168,7 +168,6 @@
 	-@if [ ! -d $(DESTDIR)$(man3dir)      ]; then mkdir -p $(DESTDIR)$(man3dir); fi
 	-@if [ ! -d $(DESTDIR)$(pkgconfigdir) ]; then mkdir -p $(DESTDIR)$(pkgconfigdir); fi
 	cp $(STATICLIB) $(DESTDIR)$(libdir)
-	cp $(SHAREDLIBV) $(DESTDIR)$(sharedlibdir)
 	cd $(DESTDIR)$(libdir); chmod u=rw,go=r $(STATICLIB)
 	-@(cd $(DESTDIR)$(libdir); $(RANLIB) libz.a || true) >/dev/null 2>&1
 	-@cd $(DESTDIR)$(sharedlibdir); if test "$(SHAREDLIBV)" -a -f $(SHAREDLIBV); then \
diff -burN orig.zlib-1.2.5/configure zlib-1.2.5/configure
--- orig.zlib-1.2.5/configure	2010-04-20 00:15:19 -0400
+++ zlib-1.2.5/configure	2011-03-06 01:11:35 -0500
@@ -124,8 +124,8 @@
   MINGW*|mingw*)
 # temporary bypass
         rm -f $test.[co] $test $test$shared_ext
-        echo "Please use win32/Makefile.gcc instead."
-        exit 1
+#        echo "Please use win32/Makefile.gcc instead."
+#        exit 1
         LDSHARED=${LDSHARED-"$cc -shared"}
         LDSHAREDLIBC=""
         EXE='.exe' ;;
